mlrun/runtimes/kubejob.py

```diff
@@ -369,20 +337,13 @@ def _run(self, runobj: RunObject, execution):
         )
         pod = client.V1Pod(metadata=new_meta, spec=pod_spec)
         try:
-            pod_name, namespace = k8s.create_pod(pod)
+            pod_name, namespace = self._get_k8s().create_pod(pod)
         except ApiException as exc:
             raise RunError(err_to_str(exc))
 
-        if pod_name and self.kfp:
-            writer = AsyncLogWriter(self._db_conn, runobj)
-            status = k8s.watch(pod_name, namespace, writer=writer)
-
-            if status in ["failed", "error"]:
-                raise RunError(f"pod exited with {status}, check logs")
```

Comment:
I'm assuming you deleted this because it's unreachable, correct? but I'm not sure why. should we deprecate the kfp param?

Reply:
it is still used in other parts of the code